yarn-project/end-to-end/src/e2e_keys.test.ts

```diff
@@ -0,0 +1,128 @@
+import { createAccounts } from '@aztec/accounts/testing';
+import {
+  type AccountWallet,
+  type AztecAddress,
+  type AztecNode,
+  Fr,
+  type L2Block,
+  type PXE,
+  type Wallet,
+} from '@aztec/aztec.js';
+import {
+  GeneratorIndex,
+  INITIAL_L2_BLOCK_NUM,
+  computeAppNullifierSecretKey,
+  computeAppSecretKey,
+  deriveMasterNullifierSecretKey,
+  deriveMasterOutgoingViewingSecretKey,
+  derivePublicKeyFromSecretKey,
+} from '@aztec/circuits.js';
+import { siloNullifier } from '@aztec/circuits.js/hash';
+import { poseidon2Hash } from '@aztec/foundation/crypto';
+import { TestContract } from '@aztec/noir-contracts.js';
+
+import { jest } from '@jest/globals';
+
+import { setup } from './fixtures/utils.js';
+
+const TIMEOUT = 120_000;
+
+describe('Key Registry', () => {
+  jest.setTimeout(TIMEOUT);
+
+  let aztecNode: AztecNode;
+  let pxe: PXE;
+  let teardown: () => Promise<void>;
+
+  let testContract: TestContract;
+
+  const secret = Fr.random();
+  let account: AccountWallet;
+
+  beforeAll(async () => {
+    let wallet: Wallet;
+    ({ aztecNode, pxe, teardown, wallet } = await setup(2));
+    testContract = await TestContract.deploy(wallet).send().deployed();
+
+    [account] = await createAccounts(pxe, 1, [secret]);
+  });
+
+  afterAll(() => teardown());
+
+  describe('using nsk_app to detect nullification', () => {
```

Comment:
This test case is copied over from `e2e_key_registry.test.ts`. The only change here is the comment bellow which I updated because of messages Lasse sent me on slack.